### PR TITLE
DYN-7693: Set geom scaling button setting to scale factor of current workspace model

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

There's a bug with the geometry scale UI button (in the top-right corner of the canvas) where it doesn't always reflect the correct scale factor setting in a workspace. It can be reproduced specifically on a fresh launch of Dynamo and upon opening a graph with a non-default scale factor setting (other than medium). This PR attempts to fix the issue.

scale_factor_demo

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Fixed a bug with the geometry scale UI button where it doesn't always reflect the correct scale factor setting in a workspace. 

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs [hunk 1]
```diff
@@ -606,7 +606,7 @@ public WorkspaceViewModel(WorkspaceModel model, DynamoViewModel dynamoViewModel)
             };
 
             geoScalingViewModel = new GeometryScalingViewModel(this.DynamoViewModel);
-            geoScalingViewModel.ScaleValue = this.DynamoViewModel.ScaleFactorLog;
+            geoScalingViewModel.ScaleValue = Convert.ToInt32(Math.Log10(Model.ScaleFactor));
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs:609]: hmm, was trying to follow the conversation on slack, do you think a unit test would help in this case?

Comment (aparajit-pratap) [reply src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs:609]: Yes, I'm working on it :)

**Comment by aparajit-pratap** [comment]
`Dynamo.Tests.ExcelTests+DSEvaluationModelTest.TestExcel` is failing. I don't believe it's related.

File: test/core/PolycurveByPoints.dyn [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  "Linting": {
+    "activeLinter": "None",
+    "activeLinterId": "7b75fb44-43fd-4631-a878-29f4d5d8399a",
+    "warningCount": 0,
+    "errorCount": 0
+  },
+  "Bindings": [],
+  "View": {
+    "Dynamo": {
+      "ScaleFactor": 10000.0,
```
Comment (aparajit-pratap) [inline test/core/PolycurveByPoints.dyn:154]: ☝️

**Review by QilongTang — APPROVED** [review]
LGTM, would you add a screen shot of the UX diff. I believe the controls on canvas would reflect the scaleFactor better is the change right?

**Comment by mjkkirschner** [comment]
hey @aparajit-pratap can you explain what this change does?

**Comment by brianAboye** [comment]
The geometry scale UI button reverts to the default setting every time I open the workflow. Is there a way of saving it with the dyn file. The geometry scale affects some of the functionality like Geometry.Intersect, so it is important to have the correct initial setting. 

Is there a way of setting the geometry scale from the API?

**Pull request merged**